developer/src/kmldmlc/src/keyman/ldml-keyboard/ldml-keyboard-xml-reader.ts

```diff
@@ -0,0 +1,51 @@
+import * as xml2js from 'xml2js';
+import LDMLKeyboardXMLSourceFile from './ldml-keyboard-xml';
+import CompilerCallbacks from '../compiler/callbacks';
+
+export default class LDMLKeyboardXMLSourceFileReader {
+
+  //private callbacks: CompilerCallbacks;
+  constructor (callbacks: CompilerCallbacks) {
+    //this.callbacks = callbacks;
+  }
+
+  /**
+   * xml2js will not place single-entry objects into arrays.
```

Comment:
it's trying to 'help' you

Reply:
To be fair, given the ambiguity inherent in XML, we either put everything into an array or we end up doing something like this. So I think this is fine :smile:

That's why I used the `explicitArray: false` parameter on line 42. If I had set that to true, I'd have had the opposite problem: having to unbox all the single-element XML nodes :grin: